src/ansys/fluent/core/solver/flobject.py

```diff
@@ -372,17 +388,49 @@ class String(SettingsBase[str], Textual):
     _state_type = str
 
 
-class Filename(SettingsBase[str], Textual):
+class InputOutputFileBase:
+    def is_input(self):
+        if self.file_purpose() == "input":
+            return True
+        return False
+
+    def before_execute(self, value):
+        try:
+            self.remote_file_handler.upload(file_name=value)
+        except AttributeError:
+            pass
+
+    def is_output(self):
+        if self.file_purpose() == "output":
```

Comment:
how are we covering in-out?

Reply:
We haven't added in-out attribute in Fluent yet. We have checked input/output for now.